60_network_learning.Rmd

```diff
@@ -0,0 +1,654 @@
+
+    5a: **"signed":** $d_{ij} = \sqrt{0.5(1-r^*_{ij})}$, where strongly negatively associated taxa have the largest distance and are placed further apart in the network.
+
+    5b: **"unsigned":** $d_{ij} = \sqrt{1-{r_{ij}^*}^2}$, resulting in a small distance between strongly associated taxa (regardless of the sign).
+
+6.  **Transformation into similarity / edge weight:** Finally, the dissimilarities are transformed into similarities by $s_{ij} = 1 - d_{ij}$, which are used as edge weights. Thus, the similarity matrix is equal to the adjacency matrix in a weighted network.
+
+In the following, we show an application for each of the three types of association: correlation, proportionality, and partial correlation. 
+
+We demonstrate the workflow using the the **PeerJ data set** [@potbhare2022skin]. It contains skin microbial profiles of 58 subjects.
+
+```{r, message=FALSE, warning=FALSE}
+library(mia)
+```
+
+```{r load_data}
+data("peerj13075", package = "mia")
+tse <- peerj13075
+dim(tse)
+```
+
+```{r, echo=FALSE}
+# The networks shown in this chapter take several minutes to generate. 
+# Therefore, the network objects have been stored in a folder and are loaded 
+# here to reduce the time needed to knit the book.
+load("general/network_data/networks.RData")
+```
+
+### Install packages
+
+Three packages used in this section are available on GitHub only: `SpiecEasi`, `SPRING`, and `NetCoMi`. We recommend that you install these packages before proceeding.
+
+```{r install_packages}
+if(!require(SpiecEasi)){
+  devtools::install_github("zdk123/SpiecEasi")
+}
+
+if(!require(SPRING)){
+  devtools::install_github("GraceYoon/SPRING")
+}
+
+if(!require(NetCoMi)){
+  devtools::install_github("stefpeschel/NetCoMi", force = TRUE,
+                           dependencies = c("Depends", "Imports", "LinkingTo"),
+                           repos = c("https://cloud.r-project.org/",
+                                     BiocManager::repositories()))
+}
+```
+
+
+### Data preparation
+
+Before we apply the network learning methods, we perform some data preparation steps similar to that in section \@ref(differential-abundance):
+
+-   Aggregate to genus level (would not be necessary here since the data is already at genus level)
+-   Prevalence filtering (keep genera with prevalence \> 20%)
+-   Add assay with clr-transformed abundances
+
+```{r preprocessing}
+# Agglomerate by genus and subset by prevalence
+tse <- subsetByPrevalentFeatures(tse,
+                                 rank = "genus",
+                                 prevalence = 0.2,
+                                 detection = 0)
```

Comment:
This does not agglomerate (although the text above says so), this filters out the rare. 

Agglomeration would instead happen with:

`tse <- mergeFeaturesByPrevalence(tse, rank = "genus", prevalence = 0.2, detection = 0)`